pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -82,20 +89,22 @@ func (fc *fluxForCluster) commitFluxUpgradeFilesToGit(ctx context.Context) error
 	config := fc.clusterSpec.GitOpsConfig
 	repository := config.Spec.Flux.Github.Repository
 
+	logger.V(3).Info("Updating repo file structure...")
+	if err := fc.updateFileStructure(); err != nil {
```

Comment:
This code won't be executed unless there is a change in the flux version. However, we want to execute it every time
I would vote to move this code to a new public method `UpdateLegacyFileStructure` (or whatever) and call it directly from the upgrade workflow (maybe from a new task to indicate that it's independent from the components upgrade?)